nexus/src/app/sagas/instance_start.rs

```diff
@@ -1116,33 +1161,45 @@ mod test {
 
         cptestctx.dendrite.insert(SwitchLocation::Switch0, new_switch0);
 
-        let dpd_client = dpd_client::Client::new(
-            &format!("http://[{addr}]:{port}"),
-            client_state,
-        );
-
         // Ensure that the nat entry for the address has made it onto the new switch0 dendrite.
-        // This might take a few seconds while the new dendrite comes online.
-        let mut nat_entries = vec![];
-        for _ in 0..5 {
-            nat_entries = dpd_client
-                .nat_ipv4_list(
-                    &std::net::Ipv4Addr::new(10, 0, 0, 0),
-                    None,
-                    None,
-                )
-                .await
-                .unwrap()
-                .items
-                .clone();
-
-            if !nat_entries.is_empty() {
-                break;
-            }
-            sleep(tokio::time::Duration::from_secs(1)).await;
-        }
-
-        assert_eq!(nat_entries.len(), 1);
+        // This might take some time while the new dendrite comes online.
+        let poll_interval = Duration::from_secs(1);
+        let poll_max = Duration::from_secs(60);
+
+        poll::wait_for_condition(
+            move || {
+                let dpd_client = dpd_client::Client::new(
+                    &format!("http://[{addr}]:{port}"),
+                    client_state.clone(),
+                );
+
+                async move {
+                    let result = match dpd_client
+                        .nat_ipv4_list(
+                            &std::net::Ipv4Addr::new(10, 0, 0, 0),
+                            None,
+                            None,
+                        )
+                        .await
+                    {
+                        Ok(v) => Ok(v),
+                        // sometimes we reach this point before the dendrite API is available
+                        Err(_) => Err(poll::CondCheckError::<()>::NotYet),
+                    }?;
+
+                    match result.items.is_empty() {
+                        // if the nat entries are empty the dendrite NAT RPW hasn't
+                        // successfully reconciled yet
+                        true => Err(poll::CondCheckError::<()>::NotYet),
+                        false => Ok(()),
+                    }
```

Comment:
```suggestion
                    if result.items.is_empty() {
                        // if the nat entries are empty the dendrite NAT RPW hasn't
                        // successfully reconciled yet
                        Err(poll::CondCheckError::<()>::NotYet)
                    } else {
                        Ok(())
                    }
```

(was there a reason this was a `match`?)